### PR TITLE
Vehicle speed switching to frequency sensor #3106

### PR DESCRIPTION
**Pull request by alxrMironov**

### CODE REVIEW

File: firmware/controllers/sensors/converters/vehicle_speed_converter.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class VehicleSpeedConverter : public SensorConverter {
+public:
+	DECLARE_ENGINE_PTR;
+
+	VehicleSpeedConverter() {
+
+	}
+
+	SensorResult 
+	convert(float frequency) const override {
```
Comment (mck1117) [inline firmware/controllers/sensors/converters/vehicle_speed_converter.h:13]: format

File: firmware/controllers/sensors/converters/vehicle_speed_converter.h [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#include "pch.h"
+#include "sensor_converter_func.h"
+
+class VehicleSpeedConverter : public SensorConverter {
+public:
+	DECLARE_ENGINE_PTR;
+
+	VehicleSpeedConverter() {
```
Comment (mck1117) [inline firmware/controllers/sensors/converters/vehicle_speed_converter.h:8]: no need for empty constructor, this is what c++ will generate by default

File: firmware/controllers/sensors/converters/vehicle_speed_converter.h (hunk 1 above)
Comment (alxrMironov) [reply firmware/controllers/sensors/converters/vehicle_speed_converter.h]: Noted

File: firmware/controllers/sensors/converters/vehicle_speed_converter.h (hunk 2 above)
Comment (alxrMironov) [reply firmware/controllers/sensors/converters/vehicle_speed_converter.h]: Noted

File: firmware/init/sensor/init_vehicle_speed_sensor.cpp [hunk 3]
```diff
@@ -2,25 +2,24 @@
...
-struct GetVehicleSpeedWrapper {
-	DECLARE_ENGINE_PTR;
 
-	float getVehicleSpeed() {
-		return ::getVehicleSpeed(PASS_ENGINE_PARAMETER_SIGNATURE);
-	}
-};
 
-static GetVehicleSpeedWrapper vehicleSpeedWrapper;
+void initVehicleSpeedSensor(DECLARE_CONFIG_PARAMETER_SIGNATURE) {
```
Comment (mck1117) [inline firmware/init/sensor/init_vehicle_speed_sensor.cpp:13]: you will need an `INJECT_ENGINE_REFERENCE(&vehicleSpeedConverter);` here somewhere

Comment (alxrMironov) [reply firmware/init/sensor/init_vehicle_speed_sensor.cpp]: Noted

**Comment by mck1117** [comment]
Try removing this line:

image

We don't really care about some random canned configuration for a Citroën.

**Comment by mck1117** [comment]
looks like the same in `dodge_neon.cpp`

image

**Comment by alxrMironov** [comment]
Eh.. that explain some things.  Ok but now it may failed in other place due of trash commits and reverts.

**Comment by mck1117** [comment]
I don't mind those reverts/etc. The diff looks fine to me, just some other old configurations using incompatible pins

**Comment by alxrMironov** [comment]
Im afraid - something doesnt been reverted. =) But forget, im check it.

**Comment by mck1117** [comment]
The diff all looks ok to me, I don't think anything was left in.

**Review by mck1117 — APPROVED** [review]

File: firmware/hw_layer/hardware.cpp [hunk 4]
```diff
@@ -390,9 +390,9 @@ void applyNewHardwareSettings(DECLARE_ENGINE_PARAMETER_SIGNATURE) {
 	}
 #endif
 
-#if EFI_VEHICLE_SPEED && ! EFI_UNIT_TEST
+/* #if EFI_VEHICLE_SPEED && ! EFI_UNIT_TEST
```
Comment (mck1117) [inline firmware/hw_layer/hardware.cpp:393]: we can remove these fully in the next PR, don't worry about it for now to keep this PR clean.

**Pull request merged**